src/SFA.DAS.LevyTransferMatching.Web.UnitTests/Controllers/OpportunitiesControllerTests.cs

```diff
@@ -123,22 +123,85 @@ public void POST_Detail_Yes_Selected_Redirects_To_SelectAccount()
-                .Setup(x => x.GetUserEncodedAccountId())
-                .ReturnsAsync(encodedAccountId);
+                .Setup(x => x.GetSelectAccountViewModel(It.Is<SelectAccountRequest>(y => y == selectAccountRequest)))
+                .ReturnsAsync(viewModel);
 
             string encodedPledgeId = _fixture.Create<string>();
 
-            var redirectToActionResult = await _opportunitiesController.SelectAccount(encodedPledgeId) as RedirectToActionResult;
+            var actionResult = await _opportunitiesController.SelectAccount(selectAccountRequest);
+            var redirectToActionResult = actionResult as RedirectToActionResult;
 
+            Assert.IsNotNull(actionResult);
             Assert.IsNotNull(redirectToActionResult);
+            
             Assert.AreEqual(redirectToActionResult.ActionName, nameof(OpportunitiesController.Apply));
         }
 
+        [Test]
+        public async Task GET_SelectAccount_AccountsReturnedMoreThanOne_ReturnsViewWithAccounts()
+        {
+            var selectAccountRequest = _fixture.Create<SelectAccountRequest>();
+            var expectedViewModel = _fixture
+                .Build<SelectAccountViewModel>()
+                .With(x => x.Accounts, _fixture.CreateMany<SelectAccountViewModel.Account>())
+                .Create();
+
+            _orchestrator
+                .Setup(x => x.GetSelectAccountViewModel(It.Is<SelectAccountRequest>(y => y == selectAccountRequest)))
+                .ReturnsAsync(expectedViewModel);
+
+            string encodedPledgeId = _fixture.Create<string>();
+
+            var actionResult = await _opportunitiesController.SelectAccount(selectAccountRequest);
+            var viewResult = actionResult as ViewResult;
+            object model = viewResult.Model;
+            var actualViewModel = model as SelectAccountViewModel;
+
+            Assert.IsNotNull(actionResult);
+            Assert.IsNotNull(viewResult);
+            Assert.IsNotNull(model);
+            Assert.IsNotNull(actualViewModel);
+
+            Assert.AreEqual(expectedViewModel, actualViewModel);
+        }
+
+        [Test]
+        public async Task GET_SelectAccount_NoAccountsReturned_ReturnsEmptyView()
+        {
+            var selectAccountRequest = _fixture.Create<SelectAccountRequest>();
+            var expectedViewModel = _fixture
+                .Build<SelectAccountViewModel>()
+                .With(x => x.Accounts, _fixture.CreateMany<SelectAccountViewModel.Account>(0))
+                .Create();
+
+            _orchestrator
+                .Setup(x => x.GetSelectAccountViewModel(It.Is<SelectAccountRequest>(y => y == selectAccountRequest)))
+                .ReturnsAsync(expectedViewModel);
+
+            string encodedPledgeId = _fixture.Create<string>();
+
+            var actionResult = await _opportunitiesController.SelectAccount(selectAccountRequest);
+            var viewResult = actionResult as ViewResult;
+            object model = viewResult.Model;
```

Comment:
I think this was just for style, and to easily identify if a type is changed - i.e. the test will fail on line 170 if a different type of `IActionResult` is changed, or 171 if no model is returned at all (rather than `NullReferenceException`ing before getting to the `Assert`s).

Fair point though!